packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -348,6 +450,17 @@ class ContainerLayer extends Layer {
     return child == equals;
   }
 
+  @override
+  void updateSubtreeNeedsAddToScene() {
+    _subtreeNeedsAddToScene = _needsAddToScene || alwaysNeedsAddToScene;
```

Comment:
it's a pity ContainerLayer directly accesses private fields of its superclass (it means e.g. that you can't create a different implementation of ContainerLayer), but I don't immediately see a clean way to fix this, so, oh well.

Reply:
Oh, I expected Dart Analysis to give me errors if  a subclass is accessing its parent class' private fields! It seems to be not the case and I have to be careful by myself. I think `super.updateSubtreeNeedsAddToScene()` would work well in this case.